### PR TITLE
feat: Implemented Voice Search

### LINKED ISSUE #1357: Implement Voice Search
**Describe the feature you'd like**
Voice Search is implemented in the app.
proper permission has been taken.
It is a necessary feature.
**Would you like to work on the issue?**
- [x] Yes

### PR DESCRIPTION
**Pull request by somenath1435**
Fixes #1357 

Changes: Voice Search is used.
proper permission is taken.
It is a necessary feature.

Screenshots for the change:

voice

### CODE REVIEW

**Comment by somenath1435** [comment]
@nikit19 @iamareebjamal please help

**Comment by aggarwalpulkit596** [comment]
There is a wildcard import in one of you files which is causing the Travis build fail

**Comment by somenath1435** [comment]
@aggarwalpulkit596 Can you suggest a way to fix it?

**Comment by aggarwalpulkit596** [comment]
Just remove any generic import ending with .*

**Comment by ShridharGoel** [comment]
Build is failing due to format violations

**Comment by somenath1435** [comment]
@ShridharGoel I had carefully refactored the code in android studio before pushing to github... Can you suggest a way to fix the error?

**Comment by aman210697** [comment]
> @ShridharGoel I had carefully refactored the code in android studio before pushing to github... Can you suggest a way to fix the error?

 Run this command 'gradlew spotlessApply' to fix the error

**Comment by ShridharGoel** [comment]
Also, the UI needs to be better. No need to write 'Use Voice Search' just the icon is sufficient and that icon should be at the right end of the 'Enter a location' field, this is what I would suggest.

**Comment by somenath1435** [comment]
@iamareebjamal @nikit19 please review

**Comment by iamareebjamal** [comment]
UI is looking really bad. Take a look at existing apps which use voice search

**Comment by somenath1435** [comment]
Improved looks:
Screenshot_20190321-003240

**Comment by iamareebjamal** [comment]
Still, do you seriously think it looks good?

**Comment by somenath1435** [comment]
How about this?
Screenshot_20190321-051718

**Comment by liveHarshit** [comment]
Just check voice search in play store and other applications.

**Comment by somenath1435** [comment]
@liveHarshit please see this:-
Screenshot_20190323-074349

File: app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt [hunk 1]
```diff
@@ -46,6 +54,11 @@ class SearchLocationFragment : Fragment() {
             rootView.currentLocation.visibility = View.GONE
         })
 
+        rootView.voice_button.setOnClickListener {
+            checkAudioPermission()
+            speechInput()
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: Will crash if permission denied

Comment (somenath1435) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: @iamareebjamal made changes , please review

File: app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt [hunk 2]
```diff
@@ -46,6 +54,15 @@ class SearchLocationFragment : Fragment() {
             rootView.currentLocation.visibility = View.GONE
         })
 
+        rootView.voice_button.setOnClickListener {
+            Utils.hideSoftKeyboard(rootView.context, rootView)
+            checkAudioPermission()
+            if (ContextCompat.checkSelfPermission(requireContext(), Manifest.permission.RECORD_AUDIO)
+                == PackageManager.PERMISSION_GRANTED)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: Duplicate check present in checkAudioPermission method

**Comment by somenath1435** [comment]
@iamareebjamal Used vector icon. PR is updated

**Comment by somenath1435** [comment]
@iamareebjamal Any other changes required?

**Comment by somenath1435** [comment]
@iamareebjamal I think this PR is ready.

**Comment by iamareebjamal** [comment]
As told before, ask peer review first

**Comment by somenath1435** [comment]
@liveHarshit What's your opinion?

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -16,8 +17,8 @@
         android:theme="@style/AppTheme">
         <activity
             android:name=".MainActivity"
-            android:windowSoftInputMode="adjustPan"
-            android:theme="@style/SplashTheme">
+            android:theme="@style/SplashTheme"
+            android:windowSoftInputMode="adjustPan">
```
Comment (liveHarshit) [inline app/src/main/AndroidManifest.xml]: Why these changes?

File: app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -26,6 +27,6 @@
         </activity>
         <meta-data
             android:name="com.stripe.android.API_KEY"
-            android:value="${STRIPE_API_TOKEN}"/>
+            android:value="${STRIPE_API_TOKEN}" />
```
Comment (liveHarshit) [inline app/src/main/AndroidManifest.xml]: Extra

File: app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt [hunk 5]
```diff
@@ -46,6 +54,15 @@ class SearchLocationFragment : Fragment() {
             rootView.currentLocation.visibility = View.GONE
         })
 
+        rootView.voice_button.setOnClickListener {
+            Utils.hideSoftKeyboard(rootView.context, rootView)
+            checkAudioPermission()
+            if (ContextCompat.checkSelfPermission(requireContext(), Manifest.permission.RECORD_AUDIO)
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: Why you are checking again if `checkAudioPermission()` method is already there?

**Comment by somenath1435** [comment]
@liveHarshit I'll make necessary changes and update accordingly.

File: app/src/main/AndroidManifest.xml (hunk 4 above)
Comment (somenath1435) [reply app/src/main/AndroidManifest.xml]: Since I've added RECORD_AUDIO permission in manifest, everyline went down by one.
I've checked it in my IDE.
In mine, last line is line30.
No extra space is there.

File: app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt (hunk 5 above)
Comment (somenath1435) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: Removed extra code.

File: app/src/main/AndroidManifest.xml (hunk 4 above)
Comment (somenath1435) [reply app/src/main/AndroidManifest.xml]: Removed.

**Comment by somenath1435** [comment]
@liveHarshit PR is updated

File: app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt [hunk 6]
```diff
@@ -46,6 +54,11 @@ class SearchLocationFragment : Fragment() {
             rootView.currentLocation.visibility = View.GONE
         })
 
+        rootView.voice_button.setOnClickListener {
+            Utils.hideSoftKeyboard(rootView.context, rootView)
+            checkAudioPermission()
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: What if permission is already granted?

**Comment by somenath1435** [comment]
@liveHarshit please see the code now

**Comment by somenath1435** [comment]
@iamareebjamal please review
